.github/workflows/ci.yml

```diff
@@ -28,6 +28,9 @@ jobs:
             .rdf-test-suite-cache
             .rdf-test-suite-ldf-cache
           key: ${{ runner.os }}-test-modules-${{ secrets.CACHE_VERSION }}-${{ hashFiles('**/yarn.lock') }}
+      - name: Remove node 18 incompatible dependencies, if on node 18
+        if: ${{ matrix.node-version == '18.x'}}
+        run: yarn remove @inrupt/solid-client-authn-core @inrupt/solid-client-authn-node @solid/community-server -W
```

Comment:
I would suggest adding these deps here via `yarn add` if _not_ on Node 18.
Because currently, devs cloning the Comunica repo on Node 18 and installing will see errors, which is probably not what we want.

Reply:
Mh, good point! For devs working on Node 16, the issue remaining would be the missing dependency errors when running tests, wouldn't it?.
How about, I move that functionality to the package.json `preinstall` script instead?

Reply:
> Mh, good point! For devs working on Node 16, the issue remaining would be the missing dependency errors when running tests, wouldn't it?

Ah yes, you're right.

> How about, I move that functionality to the package.json preinstall script instead?

Yes, that sounds like a clean solution!

Reply:
Ok, I added the following:
```
    "COMMENT1": "echo ATTENTION: The command preinstall will also check, if on node 18 and remove node 18-incompatible dependencies or add them, if not on node 18. TODO: Remove this, once the dependencies support node 18.",
    "preinstall": "if [ $( node -v | sed -r \"s/v([0-9]+).*/\\1/g\" ) = 18 ]; then yarn remove -W --ignore-scripts --ignore-engines @inrupt/solid-client-authn-core @inrupt/solid-client-authn-node @solid/community-server 2> /dev/null; else yarn add -DW --ignore-scripts @inrupt/solid-client-authn-core @inrupt/solid-client-authn-node @solid/community-server 2> /dev/null ;fi; node -e \"if (!fs.existsSync('packages/runner-cli/bin/run.js')) { var os = fs.createWriteStream('packages/runner-cli/bin/run.js'); os.write('#!/usr/bin/env node\\nconsole.error(\\'Temporary runner script during lerna installation. If you see this, please run yarn install again.\\');\\nprocess.exit(1);'); os.end(); }\" && node -e \"if (!fs.existsSync('packages/runner/bin/compile-config.js')) { var os = fs.createWriteStream('packages/runner/bin/compile-config.js'); os.write('#!/usr/bin/env node\\nconsole.error(\\'Temporary compile-config script during lerna installation. If you see this, please run yarn install again.\\');\\nprocess.exit(1);'); os.end(); }\"",
```

Or in a more readable form: Add or remove the dependencies depending on node version. I added the `yarn add`, to prevent missing dependency issues when switching node versions.
```
if [ $( node -v | sed -r \"s/v([0-9]+).*/\\1/g\" ) = 18 ]; then
    yarn remove -W --ignore-scripts --ignore-engines @inrupt/solid-client-authn-core @inrupt/solid-client-authn-node @solid/community-server 2> /dev/null;
else
    yarn add -DW --ignore-scripts @inrupt/solid-client-authn-core @inrupt/solid-client-authn-node @solid/community-server 2> /dev/null ;
fi; 
```

I'm not too happy with the `COMMENT1` "script", but I guess it's better than not having a hint as to what this does, especially if running in errors.

Reply:
> I'm not too happy with the COMMENT1 "script", but I guess it's better than not having a hint as to what this does, especially if running in errors.

Perhaps this could be the following:

```
"temp:remove_broken_node_18_dependencies": "if [ $( node -v | sed -r \"s/v([0-9]+).*/\\1/g\" ) = 18 ]; then yarn remove -W --ignore-scripts --ignore-engines @inrupt/solid-client-authn-core @inrupt/solid-client-authn-node @solid/community-server 2> /dev/null; else yarn add -DW --ignore-scripts @inrupt/solid-client-authn-core @inrupt/solid-client-authn-node @solid/community-server 2> /dev/null ;fi; node -e \"if (!fs.existsSync('packages/runner-cli/bin/run.js')) { var os = fs.createWriteStream('packages/runner-cli/bin/run.js'); os.write('#!/usr/bin/env node\\nconsole.error(\\'Temporary runner script during lerna installation. If you see this, please run yarn install again.\\');\\nprocess.exit(1);'); os.end(); }\" && node -e \"if (!fs.existsSync('packages/runner/bin/compile-config.js')) { var os = fs.createWriteStream('packages/runner/bin/compile-config.js'); os.write('#!/usr/bin/env node\\nconsole.error(\\'Temporary compile-config script during lerna installation. If you see this, please run yarn install again.\\');\\nprocess.exit(1);'); os.end(); }\""
"preinstall": "npm run temp:remove_broken_node_18_dependencies"
```